ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/permIdRequest/views/PermanentIdRequestViewImpl.java

```diff
@@ -0,0 +1,250 @@
+            .addSelectionChangedHandler(new SelectionChangedHandler<PermanentIdRequest>() {
+
+                @Override
+                public void onSelectionChanged(SelectionChangedEvent<PermanentIdRequest> event) {
+                    onSelectionChange(event);
+                }
+            });
+        grid.getView().setEmptyText(appearance.noRows());
+    }
+
+    @UiHandler("updateBtn")
+    void onUpdateBtnClicked(SelectEvent event) {
+        final UpdatePermanentIdRequestDialog dialog = new UpdatePermanentIdRequestDialog(grid.getSelectionModel()
+                                                                                             .getSelectedItem(),
+                                                                                         presenter,
+                                                                                         factory);
+
+        dialog.setHeadingText(appearance.updateStatus());
+        dialog.getOkButton().setText(appearance.update());
+        dialog.getOkButton().addSelectHandler(new SelectHandler() {
+
+            @Override
+            public void onSelect(SelectEvent event) {
+                final PermanentIdRequestUpdate update = dialog.getPermanentIdRequestUpdate();
+                presenter.updateRequest(update);
+            }
+        });
+
+        dialog.show();
+    }
+
+    @UiHandler("metadataBtn")
+    void onMetadataBtnClicked(SelectEvent event) {
+        presenter.fetchMetadata();
+    }
+
+    @UiHandler("createDOIBtn")
+    void onCreateDOIBtnClicked(SelectEvent event) {
+        final MessageBox amb = new MessageBox(appearance.request()
+                                                      + " "
+                                                      + grid.getSelectionModel()
+                                                            .getSelectedItem()
+                                                            .getType(),
+                                              appearance.confirmCreate(grid.getSelectionModel()
+                                                                           .getSelectedItem()
+                                                                           .getType()
+                                                                           .toString()));
+        amb.setPredefinedButtons(PredefinedButton.YES, PredefinedButton.NO, PredefinedButton.CANCEL);
+        amb.setIcon(MessageBox.ICONS.question());
+        amb.addDialogHideHandler(new DialogHideHandler() {
+            @SuppressWarnings("incomplete-switch")
+            @Override
+            public void onDialogHide(DialogHideEvent event) {
+
+                switch (event.getHideButton()) {
+                    case YES:
+                        presenter.createPermanentId();
+                        break;
+                    case NO:
+                        break;
+                    case CANCEL:
+                        break;
+                }
+            }
```

Comment:
Rather than suppressing the `incomplete-switch`, could the `NO` and `CANCEL` cases just be replaced with a `default` case, since they don't do anything?